WebHostLib/customserver.py

```diff
@@ -245,25 +245,19 @@ async def start_room(room_id):
             ctx.shutdown_task = asyncio.create_task(auto_shutdown(ctx, []))
             await ctx.shutdown_task
 
-            # ensure auto launch is on the same page in regard to room activity.
-            with db_session:
-                room: Room = Room.get(id=ctx.room_id)
-                room.last_activity = datetime.datetime.utcnow() - datetime.timedelta(seconds=room.timeout + 60)
-
         except (KeyboardInterrupt, SystemExit):
-            with db_session:
-                room = Room.get(id=room_id)
-                # ensure the Room does not spin up again on its own, minute of safety buffer
-                room.last_activity = datetime.datetime.utcnow() - datetime.timedelta(minutes=1, seconds=room.timeout)
+            pass
         except Exception:
             with db_session:
                 room = Room.get(id=room_id)
                 room.last_port = -1
-                # ensure the Room does not spin up again on its own, minute of safety buffer
-                room.last_activity = datetime.datetime.utcnow() - datetime.timedelta(minutes=1, seconds=room.timeout)
             raise
         finally:
+            with db_session:
+                # ensure the Room does not spin up again on its own, minute of safety buffer
+                room.last_activity = datetime.datetime.utcnow() - datetime.timedelta(minutes=1, seconds=room.timeout)
             logging.info(f"Shutting down room {room_id} on {name}.")
+            await asyncio.sleep(5)
```

Comment:
This seems jank, but I don't think it should make problems.

Reply:
It fixed the race condition real easy ;)